Pass trigger props to menu renderer (#65)

CarstenWickner opened the pull request:
Introducing `connectMenu()` function as per @charlie-rudenstal's proposal on Issue #65.

Allowing the dynamic rendering of a single menu based on the specific props of the `ContextMenuTrigger` that was used to display the `ContextMenu`.

This PR also includes a new example page showing the new method's usage, based on the multi target example.

CarstenWickner commented:
Since the shared `eslint-config-vkbansal` dependency is not fixed and was updated recently, I also had to fix a number of new eslint errors in existing files.

vkbansal commented on src/connectMenu.js:
```diff
@@ -0,0 +1,54 @@
+            }
+
+            render() {
+                return <Child {...this.props} context={this.state.context} />;
```
Isn't `context` confusing because react has its own `context`?

vkbansal commented:
Thanks for the PR.

Sorry for the eslint issues! I was experimenting with the new rules.
Can you please revert the eslint changes?

CarstenWickner commented:
1. Renamed 'context' to 'trigger' to avoid confusion.
2. Cleaned up eslint changes again.
3. Fixed new example (always have to render the `<ContextMenu/>`)
4. Included styling for disabled menu item in examples.

vkbansal commented on src/connectMenu.js:
```diff
@@ -0,0 +1,56 @@
+            }
+
+            render() {
+                return <Child {...this.props} trigger={this.state.trigger} />;
```
One thing I was thinking is that, the `connectMenu` is to be used only with `ContextMenu`. So why not use it here directly. @CarstenWickner What do you think?

CarstenWickner replied:
Yeah, I tried this but the structure gets all wrong.
The custom menu component (e.g. `DynamicMenu`) would return only a list of `MenuItem` on `render()` then, but react prefers only a single root element (i.e. `ContextMenu`).
I agree it would be better to avoid duplicating the menu's `id` both on the `connectMenu()` call and on the wrapped `DynamicMenu`'s root `ContextMenu`.

If there's a nice way, I cannot see it yet.

CarstenWickner commented:
I added the `id` attribute for convenience sake to avoid having to specify it inside the custom menu component (only on the `connectMenu( id )( Component )` call then.

The only option to automatically create the `<ContextMenu/>` wrapper would be to not expect an actual component as the `connectMenu()()`'s second argument but expect a function that returns a `MenuItem|SubMenu|Array.<(MenuItem|SubMenu)>`.
But in case of an array that might lead to warnings regarding missing `key` attributes.

Writing a custom menu component (including the `<ContextMenu/>`) seems more usable, especially if one wants to set more than just the `id` prop on the `<ContextMenu/>`.

I.e. I'd leave it as-is now.
